crates/noirc_frontend/src/parser/mod.rs

```diff
@@ -217,6 +217,85 @@ fn force<'a, T: 'a>(parser: impl NoirParser<T> + 'a) -> impl NoirParser<Option<T
     parser.map(Some).recover_via(empty().map(|_| None))
 }
 
+pub trait Mode: std::fmt::Debug + Clone + Copy + 'static {
```

Comment:
why is this needed?

Reply:
I added it because I need to preserve spans and the order of items. Perhaps it can be removed and just slightly modify the AST, but for now, I don't want to change the compiler.

Reply:
When you say for now, do you mean while the PR is in draft mode? If so, then I think thats okay while you flesh out the rest of the formatting code

Reply:
removed